lib/Core/Memory.h

```diff
@@ -153,7 +153,10 @@ class ObjectState {
 
   const MemoryObject *object;
 
+public:
```

Comment:
Do we really need this public? Memory.h is not encapsulated enough as it is, and this is just making the coupling between memory and executor worse.

Reply:
@kren1 Valid point.